experiments/kompanion/cmd/export/export.go

```diff
@@ -90,10 +95,110 @@ func init() {
 
 	ExportCmd.Flags().StringArrayVarP(&targetObjects, targetObjectsFlag, "", []string{}, "object name prefix to target. Targets all if empty. Can be specified multiple times.")
 	ExportCmd.Flags().StringArrayVarP(&ignoreObjects, ignoreObjectsFlag, "", []string{}, "object name prefix to ignore. Excludes nothing if empty. Can be specified multiple times.")
+
+	ExportCmd.Flags().IntVarP(&workerRountines, workerRoutinesFlag, "", 10, "Configure the number of worker rountines to export namespaces with. Defaults to 10. ")
+}
+
+var (
+	reportName string
+	resources  []*metav1.APIResource
+)
+
+// tracks the namespaces to be exported.
+// thread safe.
+type namespacesWorkLog struct {
+	mu         sync.Mutex
+	namespaces []string // will be treated as a FIFO queue
+}
+
+var workLog *namespacesWorkLog
+
+func (n *namespacesWorkLog) Done() bool {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	return len(n.namespaces) == 0
+}
+
+func (n *namespacesWorkLog) GetWork() string {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	if len(n.namespaces) == 0 {
+		return ""
+	}
+
+	workItem, remainig := n.namespaces[0], n.namespaces[1:]
```

Comment:
Nit: it's probably easier to write this out

```
workItem := n.namespaces[0]
n.namespaces = n.namespaces[1:]
```

(Also avoids the typo in remaining)